hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -546,9 +551,11 @@ public void executeOnEntriesInternal(EntryProcessor entryProcessor, Predicate pr
         try {
             super.executeOnEntriesInternal(entryProcessor, predicate, resultingKeyValuePairs);
```

Comment:
@mdumandag @ihsandemir Do you know why don't we invalidate the near cache in executeOnEntries method of `NearCachedClientMapProxy`? I am not too familiar with near cache stuff. Is what we do correct or we forgot it somehow?

Reply:
I think we are invalidating, see the `prepareResult` method in the `NearCachedClientMapProxy`. But what is missing in that method is that we are now invalidating it no matter what. I think we need to add a boolean parameter to that method and not invalidate the returned key, if the entry processor was `ReadOnly`

Reply:
So should we go ahead with the boolean approach?

Reply:
I think so because right now we are invalidating the returned keys no matter what for the methods where we use `prepareResult` in the client side proxy, contrary to what we do in the member proxy where we invalidate them only if the entry processor was not read-only. That would make the behaviors of the client and member proxies the same